db/db_impl.h

```diff
@@ -1104,9 +1105,17 @@ class DBImpl : public DB {
                                const std::vector<CompactionInputFiles>& inputs,
                                bool* sfm_bookkeeping, LogBuffer* log_buffer);
 
+  // Request compaction tasks token from compaction thread limiter.
+  // It always succeeds if force = true or limiter is disable. 
+  bool RequestCompactionToken(ColumnFamilyData* cfd, bool force, 
+                              std::unique_ptr<TaskLimiterToken>* token,
+                              LogBuffer* log_buffer);
```

Comment:
Warning of "This line contains trailing whitespace. Consider setting up your editor to automatically remove trailing whitespace, you will save time."
runs here too